container/lxd/server.go

```diff
@@ -55,17 +55,6 @@ type Server struct {
 	clock clock.Clock
 }
 
-// MaybeNewLocalServer returns a Server based on a local socket connection,
-// if running on an OS supporting LXD containers by default.
-// Otherwise a nil server is returned.
-func MaybeNewLocalServer() (*Server, error) {
-	if !HasSupport() {
```

Comment:
My only concern is that we seem to not be checking HasSupport now.
Do we still have the property that we detect if this machine *could* support lxd containers so that `juju deploy --to lxd:X` can give a nicer failure than trying and finding out late that LXD isn't supported?

Reply:
Before, we would pass a nil server when `HasSupport` returned true, and `IsInitialized` returned whether or not the server was nil.

The change to have `IsInitialised` call `HasSupport` means there's no logical change.

But I agree, we could be smarter about early detection.